src/apps/search/TypesenseSearch.tsx

```diff
@@ -46,9 +46,23 @@ const SearchProvider = (props: { children: ReactNode }) => {
 
 const TypesenseSearch = (props: { children: ReactNode }) => {
   const config = useRuntimeConfig<any>();
+
+  const typesenseConfig = useMemo(() => {
+    // Exclude the geometry field (which contains polygons) when polygon map results
+    // are disabled. This saves bandwidth and loading time caused by unnecessarily
+    // loading extremely large polygons (i.e. > 1MB in some cases).
+    if (!config.search.polygons) {
```

Comment:
I'm not sure we need to do this automatically. I think it can be up to the developer to configure these settings appropriately.